src/transformers/models/gpt2/modeling_flax_gpt2.py

```diff
@@ -381,28 +373,10 @@ def __call__(
         batch_size, sequence_length = input_ids.shape
 
         if position_ids is None:
-            if past_key_values is not None and input_ids.shape[-1] == 1:
```

Comment:
This is only required for generation and should be handled by the `generate()` method instead

Reply:
Shouldn't we leave the API accessible to users that want to do their own generate?

Reply:
Yeah that's a good point! On the other hand, in this case I think I'd prefer to force the user to pass `position_ids` instead of adding the hacky, magic code here => so maybe just raise if `position_ids` is not provided, but cache is initialized?

Reply:
Works for me!